testutil/simapp/consumer.go

```diff
@@ -0,0 +1,58 @@
+package simapp
+
+import (
+	appWasm "github.com/CosmWasm/wasmd/app"
+	evidencekeeper "github.com/cosmos/cosmos-sdk/x/evidence/keeper"
+	slashingkeeper "github.com/cosmos/cosmos-sdk/x/slashing/keeper"
+	ibctesting "github.com/cosmos/ibc-go/v3/testing"
+	appConsumer "github.com/cosmos/interchain-security/app/consumer"
+	ibcconsumerkeeper "github.com/cosmos/interchain-security/x/ccv/consumer/keeper"
+)
+
+type ConsumerChainType uint8
+
+const (
+	Minimal ConsumerChainType = iota
+	CosmWasm
+)
+
+func GetAllConsumerChainTypes() []ConsumerChainType {
+	return []ConsumerChainType{
+		Minimal,
+		CosmWasm,
+	}
+}
+
+// TODO: introduce interface that would be implemented by all consumer App structs and add this 3 methods
+func GetConsumerKeeper(app ibctesting.TestingApp) ibcconsumerkeeper.Keeper {
+	switch app.(type) {
+	case *appConsumer.App:
+		return app.(*appConsumer.App).ConsumerKeeper
+	case *appWasm.WasmApp:
+		return app.(*appWasm.WasmApp).ConsumerKeeper
+	default:
+		panic("Unknown consumer chain type!")
+	}
+}
+
+func GetSlashingKeeper(app ibctesting.TestingApp) slashingkeeper.Keeper {
```

Comment:
I guess that the `ConsumerApp` interface would get rid of both `GetSlashingKeeper` and `GetEvidenceKeeper`, right?

Reply:
Yes, all 3 methods would be part of that interface that would be implemented by consumer chain apps.